crates/nu-explore/src/events.rs

```diff
@@ -0,0 +1,51 @@
+use std::{
+    io::Result,
+    time::{Duration, Instant},
+};
+
+use crossterm::event::{poll, read, Event, KeyEvent};
+
+pub struct UIEvents {
+    tick_rate: Duration,
+}
+
+pub struct Cfg {
+    pub tick_rate: Duration,
+}
+
+impl Default for Cfg {
+    fn default() -> Cfg {
+        Cfg {
+            tick_rate: Duration::from_millis(250),
+        }
+    }
+}
+
+impl UIEvents {
+    pub fn new() -> UIEvents {
+        UIEvents::with_config(Cfg::default())
+    }
+
+    pub fn with_config(config: Cfg) -> UIEvents {
+        UIEvents {
+            tick_rate: config.tick_rate,
+        }
+    }
+
+    pub fn next(&self) -> Result<Option<KeyEvent>> {
+        let now = Instant::now();
+        match poll(self.tick_rate) {
```

Comment:
Why the need to poll? This causes the UI to redraw 4 times/s (I was trying to debug something and boy did the logs spam).

I changed it to a simple blocking read and things seem to work just fine (if not smoother - but maybe that's placebo).

Reply:
@rgwood may have an idea too since he's done some of the latest work on explore.

I figured this was just based on the tui example when maxim wrote it a few years ago. I'd be interested in seeing/trying your alternative.

Reply:
> I'd be interested in seeing/trying your alternative.

https://github.com/nushell/nushell/pull/13574